atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1217,6 +1220,7 @@ public void abort() {
                 if (hasWrites()) {
                     throwIfPreCommitRequirementsNotMet(null, getStartTimestamp());
                 }
+                transactionOutcomeMetrics.markAbort();
```

Comment:
Query: Nothing will be recorded in metrics if the pre-commit requirements are not met. Should we record such cases as an `abort`, or a `preCommitCheckFailed`?

Reply:
I think it's not unreasonable to pick either. I'll mark them as a `preCommitCheckFailed`

Reply:
We could in the future tag these failures with the context of the check failure (not doing this here though!)